### PR TITLE
Serviceip timeout error

### PR DESCRIPTION
**Pull request by hcfw007**
#181 

To consider in another way, the Promise itself is working well, so it is logical for it to resolve an Error and throw it outside the Promise.

### CODE REVIEW

**Comment by huan** [comment]
Thank you very much for improving the v0.22 branch!

As we understand that you need to maintain the v0.22 branch for bug fixes, and I'm focusing on the latest main branch code, so I'd like to suggest that, any PRs to the v0.22 branch, will be merged by me as soon as they have reached the below three requirements:

1. The PR is for the v0.22 branch
2. The PR has been approved by at least two contributors from @wechaty/juzi 
3. The PR has passed the GitHub actions (CI is green)

I hope the above working procedure can improve the performance of your maintaining work on this branch.

Please feel free to let me know if you have any questions. If no questions, then we can start following this list from now.

Have a great day!

**Review by su-chang — APPROVED** [review]

**Review by windmemory — APPROVED** [review]

**Comment by hcfw007** [comment]
@huan For some reason Su-chang's approval did not count...

**Comment by huan** [comment]
@su-chang @windmemory Thanks for the approvals!

Please feel free to submit more PRs and let's follow this merge steps in the future for the v0.22 branch.

**Comment by huan** [comment]
> @huan For some reason Su-chang's approval did not count...

@hcfw007 It's ok for that, we can just count by human.

The reason is that GitHub will only count the approvers who have the write privileges to the repo for approval counting.

**Pull request merged**
